dashboard/app/models/user.rb

```diff
@@ -902,6 +902,36 @@ def user_levels_by_level(script)
       index_by(&:level_id)
   end
 
+  # Retrieve all user levels for the designated set of users in the given
+  # script, with a single query.
+  # @param [Enumerable<User>] users
+  # @param [Script] script
+  # @return [Hash] UserLevels by user id by level id
+  # Example return value (where 1,2,3 are user ids and 101, 102 are level ids):
+  # {
+  #   1: {
+  #     101: <UserLevel ...>,
+  #     102: <UserLevel ...>
+  #   },
+  #   2: {
+  #     101: <UserLevel ...>,
+  #     102: <UserLevel ...>
+  #   },
+  #   3: {}
+  # }
+  def self.user_levels_by_user_by_level(users, script)
+    initial_hash = Hash[users.map {|user| [user.id, {}]}]
+    UserLevel.where(
+      script_id: script.id,
+      user_id: users.map(&:id)
+    ).
+      group_by(&:user_id).
+      inject(initial_hash) do |memo, (user_id, user_levels)|
+        memo[user_id] = user_levels.index_by(&:level_id)
+        memo
+      end
+  end
```

Comment:
I know we don't want to add anything else to user.rb, but this seemed like the most appropriate place for this helper, right below `user_levels_by_level`.  I'm open to other suggestions re: organization.

Reply:
I'm ambivalent. This concerns itself with several classes (User, UserLevel, Script) while not accessing any instance members of any, so it really could live anywhere. This could even be a global thing - would make sense as a hypothetical `Util#user_levels_by_user_by_level`. 

I think putting this on `User` is fine. If we were going to put it anywhere else at the moment, maybe on `UserLevel`? That's what's getting returned and that's a nice short file.